src/revparse.c

```diff
@@ -899,9 +899,17 @@ int git_revparse(
 			rstr++;
 		}
 
-		error = git_revparse_single(&revspec->from, repo, lstr);
-		if (!error)
-			error = git_revparse_single(&revspec->to, repo, rstr);
+		error = git_revparse_single(
+			&revspec->from,
+			repo,
+			*lstr == '\0' ? "HEAD" : lstr);
+
+		if (!error) {
+			error = git_revparse_single(
+				&revspec->to,
+				repo,
+				*rstr == '\0' ? "HEAD" : rstr);
+		}
```

Comment:
Oh, that's what's going on here. The paragraph you probably mean from gitrevisions(7):

```
       In these two shorthand notations, you can omit one end and let it default to HEAD.
       For example, origin.. is a shorthand for origin..HEAD and asks "What did I do
       since I forked from the origin branch?" Similarly, ..origin is a shorthand for
       HEAD..origin and asks "What did the origin do since I forked from them?" Note that
       .. would mean HEAD..HEAD which is an empty range that is both reachable and
       unreachable from HEAD.
```

So it explicitly says that one can omit one end, only. So I think your reasoning about us rejecting when both sides are empty is correct.